File: 169.majority_element/majorityElement.py
class Solution(object):
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        # moore voting algorithm
        count = 0
        for i in nums:
            if count == 0:
                major = i
            if i == major:
                count += 1
            else:
                count -= 1
        return major

File: 169.majority_element/majorityElement.go
func majorityElement(nums []int) int {
	var count, major int
	for i, ele := range nums {
		if i == 0 {
			major = ele
			count = 1
		} else if ele != major {
			count -= 1
			if count < 0 {
				major = ele
				count = 1
			}
		} else if ele == major {
			count += 1
		}
	}
	return major
}